### ISSUE TITLE
replicas list is not in the same order as kafka-topics.sh (so you don't know which is the preferred replica)

### ISSUE THREAD

**Issue by wushujames** [edenhill/kafkacat#98]
kafka-topics.sh --describe shows the replicas list as it is stored in zookeeper, which means that the preferred replica is the first one in the list.
```
$ ./bin/kafka-topics.sh --zookeeper zk.example.com --topic topic-name --describe
Topic:topic-name	PartitionCount:5	ReplicationFactor:2	Configs:retention.ms=60480000,cleanup.policy=delete
	Topic: topic-name	Partition: 0	Leader: 4	Replicas: 4,2	Isr: 4,2
	Topic: topic-name	Partition: 1	Leader: 5	Replicas: 5,3	Isr: 5,3
	Topic: topic-name	Partition: 2	Leader: 1	Replicas: 1,4	Isr: 1,4
	Topic: topic-name	Partition: 3	Leader: 2	Replicas: 2,5	Isr: 2,5
	Topic: topic-name	Partition: 4	Leader: 3	Replicas: 3,1	Isr: 3,1
```

kafkacat is not showing them in the same order.
```
$ kafkacat -L -b kafka.example.com -t topic-name
<snip>
  topic "topic-name" with 5 partitions:
    partition 0, leader 4, replicas: 2,4, isrs: 4,2
    partition 1, leader 5, replicas: 5,3, isrs: 5,3
    partition 2, leader 1, replicas: 1,4, isrs: 1,4
    partition 3, leader 2, replicas: 5,2, isrs: 2,5
    partition 4, leader 3, replicas: 1,3, isrs: 3,1
```

Does the broker actually preserve the ordering of the replica list as it is in zookeeper, when returning metadata to a client?

**Comment by edenhill** [comment]
The metadata is displayed exactly in the order it is reported by the broker.

**Comment by wushujames** [comment]
Okay, so it's definitely not kafkacat. I don't know if the broker promises that the replicas list that comes back via a Metadata request is in the same order as what's in zookeeper. But I'll bring it up on the Kafka JIRA. Thanks! Closing this issue.

**Comment by wushujames** [comment]
FYI, this issue was fixed on the Kafka side in 0.11.0.0. The Kafka JIRA is https://issues.apache.org/jira/browse/KAFKA-5329
